### ISSUE TITLE
Better dual-stack support

### ISSUE THREAD

**Issue by jaymzh** [fluent/fluentd#2902]
**Is your feature request related to a problem? Please describe.**

It's not possible to listen on both v4 and v6 of localhost. Unlike `::` which is a somewhat-magical setting that enables both families in Linux (and all addresses), for anything else (like localhost), there's no good solution.

You can  set `bind` to `127.0.0.1` or to `::1`, but you can't set it to both addresses, or to just `localhost`.

**Describe the solution you'd like**

* When specifying a hostname like `localhost`, all relevant addresses are listend/bound. For example:

```ruby
irb(main):016:0> Addrinfo.getaddrinfo('localhost', 5140).select { |x| x.socktype == Socket::SOCK_DGRAM }           
=> [#<Addrinfo: [::1]:5140 UDP (localhost)>, #<Addrinfo: 127.0.0.1:5140 UDP (localhost)>]
```

**Describe alternatives you've considered**

An alternative might be multiple bind statements, but... that's suboptimal.

**Comment by jaymzh** [comment]
well I take that back... `::` on most linuxes is both even if it's a UDP6 socket, so you're covered for the ANY address. The localhost issue is still a thing. I will update the issue

**Comment by jaymzh** [comment]
@ganmacs can you remove the `stale` label please, this is still broken - fluent still can't handle dual-stack environments.

**Comment by ashie** [comment]
It's a duplicated issue of #3001, and #3001 has more information including a workaround solution.
Should be merged to #3001.
